seed/php-sdk/basic-auth/src/Exceptions/SeedApiException.php

```diff
@@ -48,6 +48,6 @@ public function __toString(): string
         if (empty($this->body)) {
             return "$this->message; Status Code: $this->code\n";
         }
-        return "$this->message; Status Code: $this->code; Body: $this->body\n";
+        return "$this->message; Status Code: $this->code; Body: " . json_encode($this->body) . "\n";
```

Comment:
in this case, isn't the body already json-encoded, so this would double encode?
```
        throw new SeedApiException(
            message: 'API request failed',
            statusCode: $statusCode,
            body: $response->getBody()->getContents(),
        );
        ```
In this case, maybe we just want the default `toString()` of whatever the type is? and we'll eventually give our error bodies a good `toString()`?

Reply:
In most cases yes, the body will already be the `string` representation of a JSON object. However, if we type the body as `mixed` (as we should) it's simplest to call `json_encode` to support _all_ types at this point (e.g. other `Exception` classes, etc).

It doesn't look like there's a seamless way to map a `mixed` to a `string` representation, so `json_encode` felt like the most flexible approach, especially because it formats the type in a structured, human-readable way. In the case that the body is already a JSON object, the output will effectively be a no-op.

Thoughts?

Reply:
>  In the case that the body is already a JSON object, the output will effectively be a no-op
I think if we're passing in a string like `$response->getBody()->getContents()`, it's going to re-encode that (ie, add extra quotes around it)

I think our two cases for the type of mixed are `string` (direct json from the response) or some `object`, where the object will implement `toString()`. In that case, according to chatgpt, we should be safe to just do 
```
        return "$this->message; Status Code: $this->code; Body: " . $this->body . "\n";
```

let me know if i'm missing something!

![CleanShot 2024-09-23 at 13 59 42](https://github.com/user-attachments/assets/02395e41-e0cb-468b-a1a8-6eec2dd971b0)

Reply:
Ah good point on the double quoting. I like the simple concatenation approach - let's go with that!